### PR TITLE
#26 URL validation added for input fields

### PR DESCRIPTION
**Pull request by philipperenzen**
LDWizard should now inform the user when an invalid URL as input is provided.

### CODE REVIEW

**Comment by philipperenzen** [comment]
closes #26

File: src/containers/Configure/BaseIriField.tsx [hunk 1]
```diff
@@ -22,14 +24,29 @@ const BaseIriField: React.FC<Props> = ({}) => {
...
+
   return (
     <HintWrapper hint="The base IRI is used as a base for generating IRIs out of your resources">
       <TextField
         className={styles.baseIriField}
         value={baseIriTemp}
-        onChange={(event) => setValue(event.currentTarget.value)}
+        onChange={(event) => {
+          setValue(event.currentTarget.value)
+          // URL validation
```
Comment (RossCurry) [inline src/containers/Configure/BaseIriField.tsx:43]: remove comment

File: src/containers/Configure/BaseIriField.tsx [hunk 2]
```diff
@@ -22,14 +24,29 @@ const BaseIriField: React.FC<Props> = ({}) => {
...
+          validator.isURL(event.currentTarget.value) == false ? setValidationState(false) : setValidationState(true)
```
Comment (RossCurry) [inline src/containers/Configure/BaseIriField.tsx:44]: The `validator.isURL` returns a boolean value...I presume.

So you don't even need to check it's value:
`validator.isURL(event.currentTarget.value)  ? setValidationState(false) : setValidationState(true)`

You can go even further with this then as our state accepts a boolean as input:
`setValidationState(validator.isURL(event.currentTarget.value))`

So with the name change:
`setIsValidUrl(validator.isURL(event.currentTarget.value))`

File: src/containers/Configure/ClassField.tsx [hunk 3]
```diff
@@ -18,17 +18,28 @@ interface Props {}
...
   const [transformationConfig, setTransformationConfig] = useRecoilState(transformationConfigState);
   const prefixes = useRecoilValue(prefixState);
+  
   //   Create an intermediate value here, to stop it from re-rendering
   const [selectedClassValue, setSelectedClassValue] = React.useState<string | undefined>(undefined);
   const [writtenClassValue, setWrittenClassValue] = React.useState<string>(transformationConfig.resourceClass);
   const [autocompleteError, setAutocompleteError] = React.useState<string | undefined>();
   const [autocompleteSuggestions, setAutocompleteSuggestions] = React.useState<AutocompleteSuggestion[]>([]);
-
+  const [isValidUrl, setValidationState] = React.useState<boolean>()
```
Comment (RossCurry) [inline src/containers/Configure/ClassField.tsx:27]: check naming convention comment above

File: src/containers/Configure/ClassField.tsx [hunk 4]
```diff
@@ -18,17 +18,28 @@ interface Props {}
...
+  
+   function showHelperText(){
```
Comment (RossCurry) [inline src/containers/Configure/ClassField.tsx:29]: Please check the helperText comment above, this function should be removed. The `result` variable can be left outside of this function.

File: src/containers/Configure/ClassField.tsx [hunk 5]
```diff
@@ -94,6 +104,7 @@ const ResourceClassField: React.FC<Props> = ({}) => {
           newValueString = newValue.value;
         }
         setSelectedClassValue(newValueString);
+        setValidationState(true)
```
Comment (RossCurry) [inline src/containers/Configure/ClassField.tsx:107]: This seems odd. You are changing the validation state without having validated anything. Is this supposed to be here?

File: src/containers/Configure/ClassField.tsx [hunk 6]
```diff
@@ -104,11 +115,14 @@ const ResourceClassField: React.FC<Props> = ({}) => {
               shrink: true,
             }}
             value={writtenClassValue}
-            helperText={autocompleteError || getPrefixed(writtenClassValue, prefixes) || writtenClassValue || ""}
-            error={!!autocompleteError}
+            helperText={showHelperText()}
+            error={!!autocompleteError || isValidUrl == false}
             onChange={(event) => {
+              console.error(writtenClassValue)
```
Comment (RossCurry) [inline src/containers/Configure/ClassField.tsx:121]: Why are you logging?

File: src/containers/Configure/ClassField.tsx [hunk 7]
```diff
@@ -104,11 +115,14 @@ const ResourceClassField: React.FC<Props> = ({}) => {
...
               setSelectedClassValue(undefined);
-              const prefixInfo = getPrefixInfoFromPrefixedValue(event.currentTarget.value, prefixes);
+              // URL validation
```
Comment (RossCurry) [inline src/containers/Configure/ClassField.tsx:123]: remove comment

File: src/containers/Configure/ClassField.tsx [hunk 8]
```diff
@@ -104,11 +115,14 @@ const ResourceClassField: React.FC<Props> = ({}) => {
...
+              validator.isURL(event.currentTarget.value) == false ? setValidationState(false) : setValidationState(true)
```
Comment (RossCurry) [inline src/containers/Configure/ClassField.tsx:124]: Please check the other comments:
`setIsValidUrl(validator.isURL(event.currentTarget.value))`

File: src/containers/Configure/InteractiveTableHeaders.tsx [hunk 9]
```diff
@@ -89,7 +90,17 @@ const ColumnConfigDialog: React.FC<AutoCompleteProps> = ({ selectedHeader, onClo
     (selectedHeader !== undefined && transformationConfig.columnConfiguration[selectedHeader]) || undefined;
   const [propertyIri, setPropertyIri] = React.useState(selectedColumn?.propertyIri || "");
   const [selectedRefinement, setSelectedTransformation] = React.useState(selectedColumn?.columnRefinement);
-
+  const [isValidUrl, setValidationState] = React.useState<boolean>()
```
Comment (RossCurry) [inline src/containers/Configure/InteractiveTableHeaders.tsx:93]: check previous comment about the naming convention

File: src/containers/Configure/InteractiveTableHeaders.tsx [hunk 10]
```diff
@@ -89,7 +90,17 @@ const ColumnConfigDialog: React.FC<AutoCompleteProps> = ({ selectedHeader, onClo
...
+  
+  function showHelperText(){
+    const result = autocompleteError || getPrefixed(propertyIri, prefixes) || propertyIri
```
Comment (RossCurry) [inline src/containers/Configure/InteractiveTableHeaders.tsx:96]: check previous comment about needing a better name for this variable

File: src/containers/Configure/InteractiveTableHeaders.tsx [hunk 11]
```diff
@@ -177,10 +188,13 @@ const ColumnConfigDialog: React.FC<AutoCompleteProps> = ({ selectedHeader, onClo
                     if (!newValue) return;
                     if (typeof newValue === "string") {
                       setPropertyIri(newValue);
+                      setValidationState(true)
```
Comment (RossCurry) [inline src/containers/Configure/InteractiveTableHeaders.tsx:191]: I'm not sure why you are setting the validation status here without validating anything

File: src/containers/Configure/InteractiveTableHeaders.tsx [hunk 12]
```diff
@@ -191,8 +205,8 @@ const ColumnConfigDialog: React.FC<AutoCompleteProps> = ({ selectedHeader, onClo
                         {...props}
                         autoFocus
                         label="Property URI"
-                        error={!!autocompleteError}
-                        helperText={autocompleteError || getPrefixed(propertyIri, prefixes)}
+                        error={!!autocompleteError || isValidUrl == false}
```
Comment (RossCurry) [inline src/containers/Configure/InteractiveTableHeaders.tsx:208]: As above:
`error={!!autocompleteError || !isValidUrl}`

File: src/containers/Configure/InteractiveTableHeaders.tsx [hunk 13]
```diff
@@ -201,6 +215,7 @@ const ColumnConfigDialog: React.FC<AutoCompleteProps> = ({ selectedHeader, onClo
                         inputMode="url"
                         fullWidth
                         onChange={(event) => {
+                          validator.isURL(event.currentTarget.value) == false ? setValidationState(false) : setValidationState(true)
```
Comment (RossCurry) [inline src/containers/Configure/InteractiveTableHeaders.tsx:218]: As above:
```ts
setIsValidUrl(validator.isURL(event.currentTarget.value))
```

File: src/containers/Configure/BaseIriField.tsx [hunk 14]
```diff
@@ -22,14 +24,29 @@ const BaseIriField: React.FC<Props> = ({}) => {
       return { ...state, baseIri: baseIri };
     });
   };
+  function showHelperText(){
```
Comment (RossCurry) [inline src/containers/Configure/BaseIriField.tsx:27]: Please check the `helperText` comment [below](https://github.com/pldn/LDWizard/pull/112/files#r1162879158)
Remove this function

File: src/containers/Configure/ClassField.tsx (hunk 6 above)
Comment (philipperenzen) [reply src/containers/Configure/ClassField.tsx:121]: Oops I forgot about this one, my bad!

**Comment by RossCurry** [comment]
Hey, I tested out your code and these are my comments:
testing:
1. As soon as I load a CSV file, I am greeted with an error about how my IRI is invalid, even though it should not be.
As soon as I make a change in the field the error message goes away. Seems to be with the placeholder/default value.
2. I have the same issue when I click on the advanced drop down. The placeholder is a valid URL/iri but it complains that it is an error.
3. I think the helper text would suffice to be simply 'Invalid URL'. You already have the URL being displayed in the text input.
4. Next button works even with an Invalid URL. This should probably be disabled until the URL is valid.
5. Once I have loaded in the data. Clicking on a column header brings up a modal menu with an autosuggest. If I click outside of that autocomplete to close it, the empty field complains about " being an invalid URL. I think if we haven't touced the field and it is empty, that shouldn't be an Invalid URL message, but a field/value requiered message.

**Pull request merged**